File: ios/Classes/PhoneEntryChannelConstants.swift
import Foundation

enum PhoneEntryChannelConstants {
    static let viewType = "phone_entry_view"
    static let phoneEntryBase = "hovr_native_platform_view/phone_entry"

    static func channelName(viewId: Int64) -> String {
        "\(phoneEntryBase)_\(viewId)"
    }

    static let onPhoneSubmitted = "onPhoneSubmitted"
    static let updateSubmissionStatus = "updateSubmissionStatus"
}

struct CountryCodeItem: Codable, Identifiable {
    var id: String { code }
    let name: String
    let flag: String
    let code: String
    let dial_code: String
}

File: android/src/main/kotlin/com/ridehovr/hovr_native_platform_view/PhoneEntryChannelConstants.kt
package com.ridehovr.hovr_native_platform_view

object PhoneEntryChannelConstants {
    const val VIEW_TYPE = "phone_entry_view"
    const val PHONE_ENTRY_BASE = "hovr_native_platform_view/phone_entry"
    const val ON_PHONE_SUBMITTED = "onPhoneSubmitted"
    const val UPDATE_SUBMISSION_STATUS = "updateSubmissionStatus"

    fun channelName(viewId: Int): String = "${PHONE_ENTRY_BASE}_$viewId"
}
